packages/ag-charts-enterprise/src/features/context-menu/contextMenu.ts

```diff
@@ -33,17 +32,17 @@ export class ContextMenu extends _ModuleSupport.BaseModuleInstance implements _M
     /**
      * Extra menu actions with a label and callback.
      */
-    public extraActions: Array<ContextMenuAction> = [];
+    public extraActions: NonNullable<AgContextMenuOptions['extraActions']> = [];
```

Comment:
nit: I think we're generally trying to separate options types and modules/features types, even if it produces a bit of duplication?

Reply:
Correct.

We still do this: The internal state is stored in `ContextMenu.groups`, whereas `ContextMenu.extraActions` stores the input user options.

https://github.com/ag-grid/ag-charts/blob/546335e8cd1d67aa650974ee3e87095fc5cf4c81/packages/ag-charts-enterprise/src/features/context-menu/contextMenu.ts#L53

https://github.com/ag-grid/ag-charts/blob/546335e8cd1d67aa650974ee3e87095fc5cf4c81/packages/ag-charts-enterprise/src/features/context-menu/contextMenu.ts#L10

The `ContextMenu.groups` property contains some duplicated values like `label` and `action`, but it also contains internal meta-data like `id?` and `type`:

https://github.com/ag-grid/ag-charts/blob/1a7fe5c9c9e1af099fddf2a33d5215e3b51b98f3/packages/ag-charts-community/src/chart/interaction/contextMenuRegistry.ts#L25